File: C++/Factory/source/PizzaStore.cpp
#include "PizzaStore.h"

PizzaStore::PizzaStore(std::string type)
{
	this->type = type;
}

PizzaStore::PizzaStore(SimplePizzaFactory *factory)
{
	this->factory = factory;
}

PizzaStore::~PizzaStore()
{

}

Pizza* PizzaStore::orderPizza()
{
	Pizza *pizza;

	pizza = SimplePizzaFactory::createPizza(type);

	pizza->prepare();
	pizza->bake();
	pizza->cut();
	pizza->box();

	return pizza;
}

Pizza* PizzaStore::orderPizza(std::string type)
{
	Pizza *pizza;

	pizza = factory->createPizza(type);

	pizza->prepare();
	pizza->bake();
	pizza->cut();
	pizza->box();

	return pizza;
}


void PizzaStore::setPizzaType(std::string type)
{
	this->type = type;
}

File: java/src/DesignMode/FactoryPattern/SimpleFactory/PizzaStore.java
package DesignMode.FactoryPattern.SimpleFactory;

public class PizzaStore
{
    SimplePizzaFactory factory;

    public PizzaStore(SimplePizzaFactory factory)
    {
        this.factory = factory;
    }

    public Pizza orderPizza(String type)
    {
        Pizza pizza;

        pizza = factory.createPizza(type);
        if(pizza ==null)
        {
            System.out.println("No this type Pizza: " + type);
            return pizza;
        }

        pizza.prepare();
        pizza.bake();
        pizza.cut();
        pizza.box();

        return pizza;
    }
}